libraries/botbuilder-dialogs-adaptive-teams/src/conditions/onTeamsCardAction.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @module botbuilder-dialogs-adaptive-teams
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Expression, ExpressionParserInterface } from 'adaptive-expressions';
+import { Channels } from 'botbuilder';
+import { TurnPath } from 'botbuilder-dialogs';
+import { OnInvokeActivity } from 'botbuilder-dialogs-adaptive';
+
+/**
+ * Actions triggered when a Teams InvokeActivity is received with no activity.name.
+ */
+export class OnTeamsCardAction extends OnInvokeActivity {
+    public static $kind = 'Teams.OnCardAction';
+
+    public getExpression(parser: ExpressionParserInterface): Expression {
+        // if name is null
```

Comment:
Same comment - I don't think these are necessary as the expressions read clearly already.